dissect/target/plugins/browsers/chromium.py

```diff
@@ -154,54 +166,155 @@ def downloads(self, browser_name: str = None) -> Iterator[BrowserDownloadRecord]
+                    for extension_id in extensions.keys():
+                        manifest = extensions.get(extension_id).get("manifest")
+                        if manifest:
+                            ts_install = webkittimestamp(extensions.get(extension_id).get("first_install_time"))
+                            ts_update = webkittimestamp(extensions.get(extension_id).get("last_update_time"))
+                            name = manifest.get("name")
+                            short_name = manifest.get("short_name")
+                            description = manifest.get("description")
+                            ext_path = extensions.get(extension_id).get("path")
+                            if ext_path and self.target.os == "windows":
+                                ext_path = path.from_windows(ext_path)
+                            elif ext_path:
+                                ext_path = path.from_posix(ext_path)
+                            ext_version = manifest.get("version")
+                            ext_permissions = manifest.get("permissions")
+                            manifest_version = manifest.get("manifest_version")
 
-    def _build_userdirs(self, hist_paths: list[str]) -> list[tuple]:
-        """Join the selected browser dirs with the user home path.
+                            if manifest.get("browser_action"):
+                                default_title = manifest.get("browser_action").get("default_title")
+                            else:
+                                default_title = None
+                        else:
+                            ts_install = None
+                            ts_update = None
+                            name = None
+                            short_name = None
+                            default_title = None
+                            description = None
+                            ext_version = None
+                            ext_path = None
+                            ext_permissions = None
+                            manifest_version = None
 
-        Args:
-            hist_paths: A list with browser paths as strings.
+                        yield self.BrowserExtensionRecord(
+                            ts_install=ts_install,
+                            ts_update=ts_update,
+                            browser=browser_name,
+                            id=extension_id,
+                            name=name,
+                            short_name=short_name,
+                            default_title=default_title,
+                            description=description,
+                            version=ext_version,
+                            ext_path=ext_path,
+                            from_webstore=extensions.get(extension_id).get("from_webstore"),
+                            permissions=ext_permissions,
+                            manifest_version=manifest_version,
+                            source=str(json_file),
```

Comment:
```suggestion
                            source=json_file.as_posix(),
```
flow.record's `path` will interpret a string  according to the path rules of the system it is currently running on. `str(TargetPath)` will give back the path in a representation according to the path rules of the system that is analyzed. So when analyzing a windows target while running onder Linux, it may give back `C:\some\path`, which flow record will then see as a single file (as `\` is not a path separator in Linux).

Paths in posix representation (with a '/'), will always be correctly represented. You will miss the information that it was originally a windows path though. If you want to also capture that, you'd have to do the same as for `ext_path`:

```suggestion
                           if json_file and self.target.os == "windows":
                                json_file = path.from_windows(json_file)
                            elif json_file:
                                json_file = path.from_posix(json_file)
                                
                             ...
                             
                            source=json_file,
```